docs/modules/ROOT/pages/infra_services/data_plane_manager.adoc

```diff
@@ -35,4 +33,772 @@ Fall back mechanism from one path to the other, for example, from fast path to n
+. Fast Path
+. Rescue Path
+
+Logic:
+
+. Initially we would use fast path (grpc) to tell agent the info about next step
+. We would normally use normal path to do agent provision and nfs for big file (3GB file with gzip would cost about 3s)
+. When heart beat loss decteted (via Netdata) , data plane manager would send out rescue message to MessagingQ to let adjacent ovs do remote rescue
+
+==== Problems solution:
+
+. To solve the misorder data, put them in same queue
+. Use the tag to filter msg.
+. RocketMQ could support way more topic number with stable performance, we need to fine tune the parameter for dirty page and memory usage
+. We need to identify the detail scenario for peak scenario of 3GB deployment
+
+. Since Kafka would drop lots of performance when topic number is higher, RocketMQ is stable
+image:../../images/dpm-mq3.png[image,width=550,height=270]
+
+[Figure 4] Performance drop when topics nubmer is greater
+
+image:../../images/dpm-mq1.png[image,width=400,height=240]
+
+[Figure 5] Performance drop when topics nubmer is greater
+
+. When message body is greater than 4k , the performance would drop greatly
+image:../../images/dpm-mq2.png[image,width=400,height=240]
+
+[Figure 6] Message size and the performance drop
+
+. Here is the Neutron way to start ovs and do processing
+image:../../images/neutron-dpm.png[image,width=660,height=384]
+
+[Figure 7] Neutron way to start ovs and do processing
+
+. GRPC is not fit for transfering big file since it basically focus more on the assumption of small numbers in the data
+
+== Database Schema
+
+[cols=",,,",options="header",]
+|===
+|Vpc_states |Subnet_ states |Port_ states |security_group_ states
+| | | |
+|===
+
+State could contains more same type data as array
+
+* DataPlaneManager would deploy the goal state configuration in batch to the ACA grouply in parallel then wait for the response and do next accordingly.
+
+== REST APIs
+
+=== API Snapshot
+
+[width="100%",cols="22%,12%,50%,17%"]
+|===
+|*API Name* |*Method* |*Request*|*Response*
+
+|Get GoalState
+|GET
+|/goalstate/{goalstateid}
+|GoalState Info
+<<Get_GoalState,[sample]>>
+
+|Create Port by GoalState
+|POST
+|/goalstate/{gsinfo}
+|Goal State list
+<<Create_Port_GoalState,[sample]>>
+
+|Delete Port by GoalState
+|DELETE
+|/goalstate/{gsinfo}
+|GOAL State
+<<Delete_Port_GoalState,[sample]>>
+
+|Update GoalState
+|PUT
+|/goalstate/{goalstates}
+|GOAL State
+<<Update_GoalState,[sample]>>
+
+|Bulk Create Port by GoalState
+|POST
+|/goalstate/{gsinfo}
+|Goal State list
+<<BULK_Create_Port_GoalState,[sample]>>
+
+|Bulk Delete Port by GoalState
+|DELETE
+|/goalstate/{gsinfo}
+|GOAL State
+<<BULK_Delete_Port_GoalState,[sample]>>
+
+|===
+
+=== API Specification
+
+* Get GoalState
+* Update GoalState
+* Create Port with GoalState
+* Delete Port with GoalState
+* Bulk Create Port with GoalState
+* Bulk Delete Port with GoalState
+
+
+anchor:Get_GoalState[]
+==== (1) Get GoalState Info By GoalState Request
+
+* {blank}
++
+____
```

Comment:
You could check the format on GitHub or using IDE's Associdoc plugin like https://plugins.jetbrains.com/plugin/7391-asciidoc.

Line 188 is not recognized by Associdoc. Please refer other microservice's design doc for the format.